cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -166,10 +213,100 @@ func (s *Supervisor) loadConfig(configFile string) error {
 	return nil
 }
 
-// TODO: Implement bootstrapping https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21071
-// nolint: unparam
 func (s *Supervisor) getBootstrapInfo() (err error) {
-	s.agentVersion = "1.0.0"
+	port, err := s.findRandomPort()
+
+	if err != nil {
+		return err
+	}
+
+	supervisorPort, err := s.findRandomPort()
+
+	if err != nil {
+		return err
+	}
+
+	var cfg bytes.Buffer
+
+	s.bootstrapTemplate.Execute(&cfg, map[string]any{
+		"EndpointPort":   port,
+		"InstanceUid":    s.instanceID.String(),
+		"SupervisorPort": supervisorPort,
+	})
+
+	s.writeEffectiveConfigToFile(cfg.String(), s.effectiveConfigFilePath)
+
+	srv := server.New(s.logger.Sugar())
+
+	done := make(chan struct{}, 1)
+	var connected atomic.Bool
+
+	srv.Start(newServerSettings(flattenedSettings{
+		endpoint: fmt.Sprintf("localhost:%d", supervisorPort),
+		onConnectingFunc: func(request *http.Request) {
+			connected.Store(true)
+
+		},
+		onMessageFunc: func(_ serverTypes.Connection, message *protobufs.AgentToServer) {
+			if message.AgentDescription != nil {
+				s.agentDescription = message.AgentDescription
+				identAttr := s.agentDescription.IdentifyingAttributes
+
+				for _, attr := range identAttr {
+					switch attr.Key {
+					case semconv.AttributeServiceInstanceID:
+						if attr.Value.GetStringValue() != s.instanceID.String() {
+							s.logger.Warn(
```

Comment:
Thinking more about this. If this happens it seems to be a critical error and continuing OpAMP operation may not be the best thing. Perhaps if this happens we should stop the Collector and report and error from the Supervisor to the Server. Can be added as a TODO, does not have to be done in this PR.

Reply:
I'm fine with that, but I think we should do the same with mismatches in any other identifying attributes.

I think we may want to think about a few things, so I've created an issue to track this: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29864